proxy/http/HttpSM.cc

```diff
@@ -2439,8 +2439,12 @@ HttpSM::state_cache_open_write(int event, void *data)
   if (ua_session) {
     NetVConnection *vc = ua_session->get_netvc();
     if (vc && vc->thread != this_ethread()) {
-      pending_action = vc->thread->schedule_imm(this, event, data); // Stay on the same thread!
-      return 0;
+      if (vc->thread->is_event_type(ET_NET) || vc->thread->is_event_type(SSLNetProcessor::ET_SSL)) {
+        pending_action = vc->thread->schedule_imm(this, EVENT_INTERVAL);
```

Comment:
it looks like it will crash if you change the event there because the switch below won't ever get a valid even, won't it?